mongodump/mongodump.go

```diff
@@ -511,15 +514,21 @@ func (dump *MongoDump) DumpIntent(intent *intents.Intent, buffer resettableOutpu
 	if err != nil {
 		return err
 	}
-	findQuery := &db.DeferredQuery{Coll: session.Database(intent.DB).Collection(intent.C)}
+	intendedDB := session.Database(intent.DB)
+	coll := intendedDB.Collection(intent.C)
+	findQuery := &db.DeferredQuery{Coll: coll}
 	switch {
 	case len(dump.query) > 0:
 		findQuery.Filter = dump.query
-	case dump.OutputOptions.ViewsAsCollections || dump.InputOptions.TableScan || intent.IsSpecialCollection() || intent.IsOplog():
+	case !dump.OutputOptions.ViewsAsCollections && (dump.isWiredTiger || dump.InputOptions.TableScan):
```

Comment:
I think for readability, it might make sense to make this into a new "default" case as is the most common situation we expect.  (Making the current default into a new 'mmapv1' case.) Then the switch is:
* has a query -- use the query
* views/special/oplog -- do nothing
* ! wiredTiger -- if table scan, do nothing; else if can hint _id, do that. 
* default situation -- hint $natural